### PR TITLE
Add support for PHPUnit 6, drop PHPUnit 4

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -32,10 +32,10 @@
         "symfony/stopwatch": "^3.0 || ^4.0"
     },
     "require-dev": {
-        "johnkary/phpunit-speedtrap": "^1.1",
+        "johnkary/phpunit-speedtrap": "^1.1 || ^2.0@dev",
         "justinrainbow/json-schema": "^5.0",
         "php-cs-fixer/accessible-object": "^1.0",
-        "phpunit/phpunit": "^4.8.35 || ^5.4.3",
+        "phpunit/phpunit": "^5.7.23 || ^6.4.3",
```
Comment (kubawerlos) [inline composer.json]: Sort dependencies by the way?

Comment (keradus) [reply composer.json]: done

File: tests/Console/ConfigurationResolverTest.php [hunk 2]
```diff
@@ -988,8 +981,10 @@ public function testResolveRulesWithOption()
 
     public function testResolveRulesWithUnknownRules()
     {
-        $this->setExpectedException(
-            \PhpCsFixer\ConfigurationException\InvalidConfigurationException::class,
+        $this->expectException(
+            \PhpCsFixer\ConfigurationException\InvalidConfigurationException::class
```
Comment (localheinz) [inline tests/Console/ConfigurationResolverTest.php]: Quick note: it appears that `PhpCsFixer\ConfigurationException\InvalidConfigurationException` has already been imported, so while this out of scope of the changes suggested in this PR, what do you think about simplifying the FQCN here, as well as putting the invocation on a single line?

Comment (keradus) [reply tests/Console/ConfigurationResolverTest.php]: doing it in this one place is pointless, we already have a future request to import all usages like here, that would fix issue globally

File: .appveyor.yml [hunk 3]
```diff
@@ -5,7 +5,6 @@ clone_folder: C:\projects\php-cs-fixer
 
 environment:
     matrix:
-        - php_ver: 7.1.2
```
Comment (julienfalque) [inline .appveyor.yml]: What was wrong with this version?

**Comment by julienfalque** [comment]
Arf, already merged, but 👍 anyway :)

File: .appveyor.yml (hunk 3 above)
Comment (keradus) [reply .appveyor.yml]: it was freezing on phpunit execution

Comment (keradus) [reply .appveyor.yml]: ref https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3188

**Comment by keradus** [comment]
done by #3201

**Pull request closed without merging**